cmd/utils/flags.go

```diff
@@ -660,6 +670,14 @@ var (
 		Usage: "Default minimum difference between two consecutive block's timestamps in seconds",
 		Value: eth.DefaultConfig.Istanbul.BlockPeriod,
 	}
+	IstanbulProxiedFlag = cli.BoolFlag{
+		Name:  "istanbul.proxied",
+		Usage: fmt.Sprintf("Specifies whether this node will be proxied by sentry nodes. This option requires that --%s is also used.  It will also disables discovery.", MiningEnabledFlag.Name),
+	}
+	IstanbulSentriesFlag = cli.StringFlag{
+		Name:  "istanbul.sentries",
+		Usage: fmt.Sprintf("Comma separated enode URL pairs for proxied validators"),
```

Comment:
Ya, great question.

So the sentry is going to have two enodes (one for the external network interface and one for the internal one).

Proxied validator will obviously need to know the internal one to connect to, but it also will need the know the external one when it sends (and signs) the announce message.

In theory, it's possible to modify the Istanbul communication protocol (or perhaps the discovery protocol) so that the external URL is automatically shared from the sentry to that validator.  But I felt that could be a follow up feature.

Reply:
Makes total sense, can you document this somewhere if it isn't already?